packages/flutter/lib/src/widgets/actions.dart

```diff
@@ -481,6 +481,37 @@ class CallbackAction<T extends Intent> extends Action<T> {
   Object? invoke(covariant T intent) => onInvoke(intent);
 }
 
+/// A registry of actions that is checked by the [Actions] widget
+/// for actions to be invoked in addition to its own [Actions.actions].
+///
+/// Assign one of these to the [Actions.registry] attribute of the [Actions]
+/// widget you wish to serve as the highest level in the widget hierarchy that
+/// will be searched for the actions of its subtree.
+///
```

Comment:
Document the expected lifecycle of this? Presumably, it shouldn't be recreated on every build and be long-lived (i.e. owned by a State object).

Reply:
To avoid the lifecycle issues, could this just be its own (stateful) widget that inserts an Actions widget into the tree with the actions that have been registered to it?

Reply:
Made the `ActionsRegistry` widget and remove this attribute.